src/pages/patch/patchModal/PatchModal.tsx

```diff
@@ -0,0 +1,228 @@
+import React, { useState } from "react";
+import { Modal } from "antd";
+import { useQuery, useMutation } from "@apollo/react-hooks";
+import styled from "@emotion/styled";
+import Button from "@leafygreen-ui/button";
+import Checkbox from "@leafygreen-ui/checkbox";
+import { Body, H2 } from "@leafygreen-ui/typography";
+import { uiColors } from "@leafygreen-ui/palette";
+import { useParams } from "react-router-dom";
+import { TreeSelect } from "components/TreeSelect";
+import { GET_PATCH_BUILD_VARIANTS } from "gql/queries/get-patch-build-variants";
+import { RESTART_PATCH } from "gql/mutations/restart-patch";
+import get from "lodash/get";
+import {
+  PatchBuildVariantsQuery,
+  PatchBuildVariantsQueryVariables,
+  RestartPatchMutation,
+  RestartPatchMutationVariables,
+} from "gql/generated/types";
+import { usePatchStatusSelect } from "hooks";
+import { useBannerDispatchContext } from "context/banners";
+import { TaskStatus } from "types/task";
+import { PatchBuildVariantAccordian } from "./PatchBuildVariantAccordian";
+
+const { gray } = uiColors;
+
+interface PatchModalProps {
+  visible: boolean;
+  onOk: () => void;
+  onCancel: () => void;
+}
+export const PatchModal: React.FC<PatchModalProps> = ({
+  visible,
+  onOk,
+  onCancel,
+}) => {
+  const { successBanner, errorBanner, clearAll } = useBannerDispatchContext();
+  const { id } = useParams<{ id: string }>();
+  const [abortInProgressTasks, setAbortInProgressTasks] = useState(false);
+  const [restartPatch, { loading: mutationLoading }] = useMutation<
+    RestartPatchMutation,
+    RestartPatchMutationVariables
+  >(RESTART_PATCH, {
+    onCompleted: () => {
+      onOk();
+      successBanner(`Successfully restarted patch: ${id}`);
+    },
+  });
+  const { data, loading } = useQuery<
+    PatchBuildVariantsQuery,
+    PatchBuildVariantsQueryVariables
+  >(GET_PATCH_BUILD_VARIANTS, {
+    variables: { patchId: id },
+  });
+  const patchBuildVariants = get(data, "patchBuildVariants");
+  const [
+    selectedTasks,
+    validStatus,
+    { toggleSelectedTask, setValidStatus },
+  ] = usePatchStatusSelect(patchBuildVariants);
+
+  const handlePatchRestart = async (e): Promise<void> => {
+    e.preventDefault();
+    clearAll();
```

Comment:
can this be renamed too to be like the rest of them? `clearAllBanners`

Reply:
Sure!

Reply:
tbh tho after using destructuring to pull out these functions for the banners i feel like i prefer the non destructured way of calling these functions. 

i think `banner.warningBanner()` or `banner.clearAllBanners()` or  even `banner.createWarning()` is so much more clearer of the intent and is easier to keep track of whats going on. Just my opinion tho. What do you think? @SupaJoon @tgrander

Reply:
this is how i intended for the context to be used so yes, i am on board with it. but as you will find with a lot of things in this regard, i do not have a strong opinion because my opinions are generally spent elsewhere.